### DISCUSSION TITLE
Klimerko Firmware v2.1.0

### DISCUSSION THREAD

**Discussion by epiller** [DesconBelgrade/Klimerko#37] [Announcements]
This update brings a highly requested feature: User-configurable temperature offset.
Previously, if you wanted to change the temperature offset, you'd need to do it in source code and then recompile and re-upload it. 
This time around, all you need to do is enter **WiFi Configuration Mode** and input your desired temperature offset.  
Not only that, but **humidity is now compensated for the temperature offset**, which means that humidity sensor readings are much more accurate now.

# Configurable Temperature Offset
- Configured via [WiFi Configuration Mode](/README.md/#configuring-klimerko-credentials-and-temperature-offset)
- Humidity readings are compensated for the temperature offset, which results in more accurate humidity readings than before
- Your custom temperature offset is now stored in device memory, which means it persists over firmware updates - same as device credentials
- A [factory reset](https://github.com/DesconBelgrade/Klimerko#factory-reset-your-klimerko) resets the temperature offset value as well
- Current temperature offset is also published as an asset to your AllThingsTalk Maker account
  - Make sure to create a `Sensor` asset of type `Number` named `temperature-offset` in order to use this feature
- Default temperature offset is `-4`°C, same as before
- Decimal values are supported (e.g. `-2.4`, `0.2`, `-5.12`)
- Maximum positive offset is `+25`°C and maximum negative offset is `-25`°C
- Use value `0` to disable offset
- Current temperature offset is always shown in WiFi Configuration Portal (as opposed to credentials, which aren't)

# Other changes
- Updated WiFiManager
- Adds units when displaying sensor values via serial monitor
- Shows your input values for Device Token and Device ID via Serial output if the values you wrote are disregarded for any reason
- Informs you via Serial output if there are no WiFi credentials stored on the device
- Renamed some functions

<hr /><em>This discussion was created from the release <a href='https://github.com/DesconBelgrade/Klimerko/releases/tag/2.1.0'>Klimerko Firmware v2.1.0</a>.</em>

**Comment by YuMERA** [comment]
Zašto, je sve u vezi sa ovim projektom na engleskom jeziku? Ovaj projekat je namenjen za podrucje Republike Srbije i nema nikakve veze sa stranim jezikom. Delite klimerka po prijavi zainteresovanih korisnika bez da korisnik ima bilo kakvo predznanje o elektronici ili programiranju i onda im sve to lepo objasnite na engleskom  jeziku kako da sklope ili isprogramiraju uređaj . Ja ne vidim ni jedno logično objasnjenje na ovu moju primedbu sem da se neko pravi pametan pa "priča" na engleskom, a samim tim ga je i puno briga za celukupnu zajednicu korisnika za koju teži da je što veća.

**Reply by epiller** [reply]
Klimerko je pokrenut na [Descon-u](https://descon.me) koji je medjunarodni dogadjaj na Engleskom jeziku tako da originalno nije pravljen samo za Srbiju. Ako postoji neko ko bi htio da prevede dokumentaciju i uradi pull request, to bi bilo super.
